presto-parser/src/main/java/io/prestosql/sql/parser/AstBuilder.java

```diff
@@ -1558,6 +1560,7 @@ public Node visitFunctionCall(SqlBaseParser.FunctionCallContext context)
                 filter,
                 orderBy,
                 distinct,
+                ignoreNulls,
```

Comment:
Please notice that the “RESPECT / IGNORE NULLS” argument is allowed by the grammar for every function call of the form `qualifiedName '(' (setQuantifier? expression (',' expression)*)? … `, which applies not only to window value functions for which it is dedicated.
Make sure that every invalid usage of RESPECT / IGNORE NULLS argument that has passed through `AstBuilder` is caught and fails with `NOT SUPPORTED`. Add appropriate checks in `ExpressionAnalyzer.visitFunctionCall()`.

Reply:
The grammar only allows the null treatment clause for specific window functions, so I think we could change the grammar to only allow it for window functions:
```
qualifiedName '(' ... ')'
filter? (nullTreatment? over)?
```
Although we should still add the checks in the analyzer, in case it is passed in improperly built AST.